### DISCUSSION TITLE
apollo可不可以解决应用有状态部分的问题

### DISCUSSION THREAD

**Discussion by tpxxwz** [apolloconfig/apollo#4094] [General]
服务里面用到了雪花算法里面有
workerId、datacenterId两个参数 需要每个服务配置不同来保证 生成的id全局唯一
应用在不部署到k8s里可以用  ip+端口及对应的workerId、datacenterId  的map  放到apollo来统一配置
由于k8s的ip是分配的每次不一样  那么该如何解决这个问题呢
apollo是否可以提供一个池子 不同的pop来获取配置随机分配一个workerId、datacenterId
应用下线了 又返回给池子

**Comment by nobodyiam** [comment]
This sounds like a consensus problem
